art/attacks/evasion/shapeshifter.py

```diff
@@ -0,0 +1,1122 @@
+# MIT License
+#
+# Copyright (C) The Adversarial Robustness Toolbox (ART) Authors 2020
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+# persons to whom the Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+# Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+# WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+# TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""
+This module implements ShapeShifter, a robust physical adversarial attack on Faster R-CNN object detector.
+
+| Paper link: https://arxiv.org/abs/1804.05810
+"""
+import logging
+from typing import List, Dict, Optional, Tuple, TYPE_CHECKING
+from collections import Callable
+
+import numpy as np
+import tensorflow as tf
+
+from art.attacks.attack import EvasionAttack
+from art.estimators.estimator import BaseEstimator, LossGradientsMixin, NeuralNetworkMixin
+from art.estimators.tensorflow import TensorFlowEstimator
+from art.estimators.object_detection.tensorflow_faster_rcnn import TensorFlowFasterRCNN
+from art.estimators.object_detection.object_detector import ObjectDetectorMixin
+
+if TYPE_CHECKING:
+    from tensorflow.python.framework.ops import Tensor
+    from tensorflow.python.training.optimizer import Optimizer
+
+logger = logging.getLogger(__name__)
+
+
+class ShapeShifter(EvasionAttack):
```

Comment:
Could you please add the import of `ShapeShifter` to `art.attacks.evasion.__init__.py`?

Reply:
Done.